robot/rospackages/src/mcu_control/scripts/serial_util.py

```diff
@@ -0,0 +1,129 @@
+# setup serial communications by searching for arm teensy if USB, or simply connecting to UART
+# baud: baudrate, type: node type
+# baudrate & type:
+# PdsNode = 9600, 'PDS'
+# ArmNode = 115200, 'arm'
+# ScienceNode = 115200, 'science'
+# RoverNode = 115200, 'Astro'
+
+node_baudrate = {
+  "pds": 9600,
+  "arm": 115200,
+  "science": 115200,
+  "rover": 115200
+}
+
+node_type = {
+  "pds": 'PDS',
+  "arm": 'arm',
+  "science": 'science',
+  "rover": 'Astro'
+}
+
+def init_serial(baud, type):
+    baudrate = baud
+    # in a perfect world, you can choose the baudrate
+    rospy.loginfo('Using %d baud by default', baudrate)
+    # in a perfect world, usb vs uart will be set by ROS params
+    usb = False; uart = True
+    myargv = rospy.myargv(argv=sys.argv)
+    if len(myargv) == 1:
+        rospy.loginfo('Using UART by default')
+    if len(myargv) > 1:
+        if myargv[1] == 'uart':
+            usb=False; uart=True
+            if type = 'PDS':
+                rospy.loginfo('Using UART and 9600 baud by default')
+            else:
+                rospy.loginfo('Using UART and 115200 baud by default')
+        elif myargv[1] == 'usb':
+            usb=True; uart=False
+            if type = 'PDS':
+                rospy.loginfo('Using USB and 9600 baud by default')
+            else:
+                rospy.loginfo('Using USB and 115200 baud by default')
+        else:
+            rospy.logerr('Incorrect argument: expecting "usb" or "uart"')
+            sys.exit(0)
+
+    global ser #make global so it can be used in other parts of the code
+
```

Comment:
from a software engineering perspective this function does a lot more than one thing. I would suggest that you split up the stuff below, trying to reuse code, into a function that you call a few times. Notice that there's almost the same code trying (5) times to do something